src/MongoDB.Driver.Core/Core/Operations/CountDocumentsOperation.cs

```diff
@@ -164,30 +175,22 @@ public ReadConcern ReadConcern
         public long Execute(IReadBinding binding, CancellationToken cancellationToken)
         {
             Ensure.IsNotNull(binding, nameof(binding));
-            using (var channelSource = binding.GetReadChannelSource(cancellationToken))
-            using (var channel = channelSource.GetChannel(cancellationToken))
-            using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
```

Comment:
We didn't really need these three lines. We can use the same binding to call Execute.